In [1]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt 
%matplotlib inline


In [2]:
data = pd.read_csv('Provisional_Death_Counts_for_Influenza__Pneumonia__and_COVID-19.csv',sep=',', index_col = 0, na_values = "na")
print (data.shape)
data.head

(10836, 15)


<bound method NDFrame.head of             Start Week    End Week  MMWRyear  MMWRweek    Group    Indicator  \
Data As Of                                                                     
11/25/2020  02/01/2020  02/01/2020      2020         5  By week  Week-ending   
11/25/2020  02/01/2020  02/01/2020      2020         5  By week  Week-ending   
11/25/2020  02/01/2020  02/01/2020      2020         5  By week  Week-ending   
11/25/2020  02/01/2020  02/01/2020      2020         5  By week  Week-ending   
11/25/2020  02/01/2020  02/01/2020      2020         5  By week  Week-ending   
...                ...         ...       ...       ...      ...          ...   
11/25/2020  11/21/2020  11/21/2020      2020        47  By week  Week-ending   
11/25/2020  11/21/2020  11/21/2020      2020        47  By week  Week-ending   
11/25/2020  11/21/2020  11/21/2020      2020        47  By week  Week-ending   
11/25/2020  11/21/2020  11/21/2020      2020        47  By week  Week-ending   
11/25/2020

In [3]:
data.describe()


MMWRyear      MMWRweek  COVID-19 Deaths  Total Deaths  \
count   10836.0  10836.000000      9114.000000   9716.000000   
mean     2020.0     26.000000       157.071758   1589.344998   
std         0.0     12.410246       707.549047   5318.890128   
min      2020.0      5.000000         0.000000      0.000000   
25%      2020.0     15.000000         0.000000     88.000000   
50%      2020.0     26.000000        13.000000    386.000000   
75%      2020.0     37.000000        78.000000   1207.250000   
max      2020.0     47.000000     17068.000000  78821.000000   

       Pneumonia Deaths  Influenza Deaths  Pneumonia or Influenza  \
count       8917.000000       8401.000000             8897.000000   
mean         161.858024          4.072372              166.812409   
std          578.399546         27.793888              593.282163   
min            0.000000          0.000000                0.000000   
25%            0.000000          0.000000                0.000000   
50%           32.000000          0.000000               34.000000   
75%          107.000000          0.000000              111.000000   
max        11972.000000        654.000000            12444.000000   

       Pneumonia, Influenza, or COVID-19 Deaths  
count                               9065.000000  
mean                                 249.488252  
std                                  954.621799  
min                                    0.000000  
25%                                   10.000000  
50%                                   44.000000  
75%                                  158.000000  
max                                21222.000000

In [4]:
data.columns

Index(['Start Week', 'End Week', 'MMWRyear', 'MMWRweek', 'Group', 'Indicator',
       'Jurisdiction', 'Age Group', 'COVID-19 Deaths', 'Total Deaths',
       'Pneumonia Deaths', 'Influenza Deaths', 'Pneumonia or Influenza',
       'Pneumonia, Influenza, or COVID-19 Deaths', 'Footnote'],
      dtype='object')

In [5]:
data.groupby("COVID-19 Deaths").size()

COVID-19 Deaths
0.0        4146
1.0           3
2.0           9
3.0           6
4.0           7
           ... 
13139.0       1
13841.0       1
15426.0       1
16162.0       1
17068.0       1
Length: 906, dtype: int64

In [6]:
data.isnull().sum().sum()

16040

In [7]:
data2=data.dropna()
data2.isnull().sum().sum()

0

In [57]:
data.groupby("Start Week").size()

#len(data.groupby("Start Week"))

43

In [8]:
data.groupby("Jurisdiction").size()

Jurisdiction
Alabama          172
Alaska           172
Arizona          172
Arkansas         172
California       172
                ... 
Virginia         172
Washington       172
West Virginia    172
Wisconsin        172
Wyoming          172
Length: 63, dtype: int64

In [41]:
data.Jurisdiction.unique()

array(['United States', 'Alabama', 'Alaska', 'Arizona', 'Arkansas',
       'California', 'Colorado', 'Connecticut', 'Delaware',
       'District of Columbia', 'Florida', 'Georgia', 'Hawaii', 'Idaho',
       'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana',
       'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota',
       'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada',
       'New Hampshire', 'New Jersey', 'New Mexico', 'New York',
       'New York City', 'North Carolina', 'North Dakota', 'Ohio',
       'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island',
       'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah',
       'Vermont', 'Virginia', 'Washington', 'West Virginia', 'Wisconsin',
       'Wyoming', 'HHS Region 1', 'HHS Region 2', 'HHS Region 3',
       'HHS Region 4', 'HHS Region 5', 'HHS Region 6', 'HHS Region 7',
       'HHS Region 8', 'HHS Region 9', 'HHS Region 10'], dtype=object)

In [60]:
data.groupby("Age Group").size()

Age Group
0-17 years     2709
18-64 years    2709
65+ years      2709
All Ages       2709
dtype: int64

Dividindo 2709 (quantas vezes aparece cada age group) por 63 ( o número de estados (apesar de serem menos os estados dos estados unidos (52), temos 1 United States e 10 HHS Region) obtemos o número de semanas que pode ser confirmado por ```
len(data.groupby("Start Week"))
```

In [61]:
data.groupby("Age Group").count()

Start Week  End Week  MMWRyear  MMWRweek  Group  Indicator  \
Age Group                                                                 
0-17 years         2709      2709      2709      2709   2709       2709   
18-64 years        2709      2709      2709      2709   2709       2709   
65+ years          2709      2709      2709      2709   2709       2709   
All Ages           2709      2709      2709      2709   2709       2709   

             Jurisdiction  COVID-19 Deaths  Total Deaths  Pneumonia Deaths  \
Age Group                                                                    
0-17 years           2709             2528          1616              2168   
18-64 years          2709             1942          2692              1852   
65+ years            2709             2314          2703              2412   
All Ages             2709             2330          2705              2485   

             Influenza Deaths  Pneumonia or Influenza  \
Age Group                                               
0-17 years               2543                    2100   
18-64 years              2075                    1883   
65+ years                1899                    2422   
All Ages                 1884                    2492   

             Pneumonia, Influenza, or COVID-19 Deaths  Footnote  
Age Group                                                        
0-17 years                                       2015      1685  
18-64 years                                      2003      1566  
65+ years                                        2494      1240  
All Ages                                         2553      1211